applications/asset_tracker_v2/tests/ui_module/src/ui_module_test.c

```diff
@@ -0,0 +1,401 @@
+ */
+extern int unity_main(void);
+
+/* Suite teardown finalizes with mandatory call to generic_suiteTearDown. */
+extern int generic_suiteTearDown(int num_failures);
+
+int test_suiteTearDown(int num_failures)
+{
+	return generic_suiteTearDown(num_failures);
+}
+
+void setUp(void)
+{
+	mock_modules_common_Init();
+	mock_event_manager_Init();
+
+	/* Reset internal module states. */
+	state = 0;
+	sub_state = 0;
+	sub_sub_state = 0;
+
+	/* Clear internal LED pattern transition list. */
+	transition_list_clear();
+}
+
+void tearDown(void)
+{
+	mock_modules_common_Verify();
+	mock_event_manager_Verify();
+}
+
+/* Handler that validates events sent from the UI module. */
+static void validate_ui_evt(struct event_header *eh, int no_of_calls)
+{
+	struct ui_module_event *event = cast_ui_module_event(eh);
+
+	TEST_ASSERT_EQUAL(UI_EVT_SHUTDOWN_READY, event->type);
+}
+
+/* Function used to verify the internal state of the UI module. */
+static void state_verify(enum state_type state_new, enum sub_state_type sub_state_new,
+			 enum sub_sub_state_type sub_sub_state_new)
+{
+	TEST_ASSERT_EQUAL(state_new, state);
+	TEST_ASSERT_EQUAL(sub_state_new, sub_state);
+	TEST_ASSERT_EQUAL(sub_sub_state_new, sub_sub_state);
+}
+
+/**
+ * @brief Function that validates the next node in the led pattern_transition_list.
+ *	  If the passed in led_state and duration_sec variables are 0 the function returns
+ *	  true if no more nodes are present in the list.
+ *
+ * @param led_state LED state that will be verified against.
+ * @param duration_sec Duration of the LED state.
+ *
+ * @return true if the LED state and duration is associated with the next node.
+ * @return false if the LED state and duration is not associated with the next node.
+ *
+ */
+static bool list_node_verify_next(enum led_state led_state, int16_t duration_sec)
+{
+	struct led_pattern *led_pattern = NULL;
+	sys_snode_t *node = sys_slist_get(&pattern_transition_list);
+
+	if (node == NULL && led_state == 0 && duration_sec == 0) {
+		return true;
+	} else if (node == NULL) {
+		return false;
+	}
+
+	led_pattern = CONTAINER_OF(node, struct led_pattern, header);
+
+	/* Verify the expected LED pattern. */
+	if ((led_pattern->led_state == led_state) && (led_pattern->duration_sec == duration_sec)) {
+		return true;
+	}
+
+	return false;
+}
+
+/* Function used to verify LED pattern transitions upon publication to cloud. */
+static void verify_publication(bool active_mode)
+{
+	struct data_module_event *data_module_event;
+	struct gnss_module_event *gnss_module_event;
+	enum led_state led_state_mode_verify = LED_STATE_ACTIVE_MODE;
+	enum sub_state_type sub_state_verify = SUB_STATE_ACTIVE;
+
+	if (!active_mode) {
+		led_state_mode_verify = LED_STATE_PASSIVE_MODE;
+		sub_state_verify = SUB_STATE_PASSIVE;
+	}
+
+	/* Send cloud publication event. */
+	TEST_SEND_EVENT(data, DATA_EVT_DATA_SEND, data_module_event);
+	TEST_ASSERT_EQUAL(1, list_node_verify_next(LED_STATE_CLOUD_PUBLISHING, 5));
```

Comment:
`TEST_ASSERT_TRUE` can be used here and other occurrences

Reply:
Thanks, will update 👍